zone/spell_effects.cpp

```diff
@@ -5567,6 +5620,91 @@ int32 Mob::CalcFocusEffect(focusType type, uint16 focus_id, uint16 spell_id, boo
+					Usage example: [EXCLUDE] Do not focus spell if it is from an item click from a helmet slot. [base= -1000, limit= -1, max= 4]
+					Usage example: [EXCLUDE] Do not focus spell if it is from an item click and is a 'defense armor' item type=10. [base= -110, limit= -1, max= -1]
+
+					Note: You can apply multiple includes or excludes to a single focus spell,  using multiple SPA 415 limits in the spell. Ie. Check for clicks from ItemType 10 or 11.
+				
+				*/
+							
+				if (casting_spell_inventory_slot && casting_spell_inventory_slot != -1) {
+					if (IsClient() && casting_spell_slot == EQ::spells::CastingSlot::Item && casting_spell_inventory_slot != 0xFFFFFFFF) {
+						auto item = CastToClient()->GetInv().GetItem(casting_spell_inventory_slot);
+						if (item && item->GetItem()) {
+							//If ItemType set to < -1, then we will exclude either all Subtypes (-1000), or specific items by ItemType, SubType or Slot. See above for rules.
+							if (focus_spell.base_value[i] < -1) { //Excludes
+								bool exclude_this_item = true;
+								int tmp_itemtype = (item->GetItem()->ItemType + 100) * -1;
+								//ItemType (if set to -1000, ignore and exclude any ItemType)
+								if (focus_spell.base_value[i] < -1 && focus_spell.base_value[i] != -1000) {
+									if (focus_spell.base_value[i] != tmp_itemtype) {
+										exclude_this_item = false;
+									}
+								}
+								//SubType (if set to -1, ignore and exclude all SubTypes)
+								if (focus_spell.limit_value[i] >= 0) {
+									if (focus_spell.limit_value[i] != item->GetItem()->SubType) {
+										exclude_this_item = false;
+									}
+								}
+								//item slot bitmask (if set to -1, ignore and exclude all SubTypes)
+								if (focus_spell.max_value[i] >= 0) {
+									if (focus_spell.max_value[i] != item->GetItem()->Slots) {
+										exclude_this_item = false;
+									}
+								}
+								if (exclude_this_item) {
+									return 0;
+								}
+							}
+							else {//Includes
+								LimitInclude[IncludeExistsSEFFItemClass] = true;
+								bool include_this_item = true;
+								//ItemType (if set to -1, ignore and include any ItemType)
+								if (focus_spell.base_value[i] >= 0) {
+									if (focus_spell.base_value[i] != item->GetItem()->ItemType) {
+										include_this_item = false;
+									}
+								}
+								//SubType (if set to -1, ignore and include any SubType)
+								if (focus_spell.limit_value[i] >= 0) {
+									if (focus_spell.limit_value[i] != item->GetItem()->SubType) {
+										include_this_item = false;
+									}
+								}
+								//item slot bitmask (if set to -1, ignore and include any slot)
+								if (focus_spell.max_value[i] >= 0) {
+									if (focus_spell.max_value[i] != item->GetItem()->Slots) {
+										include_this_item = false;
+									}
+								}
+
+								if (include_this_item) {
+									LimitInclude[IncludeFoundSEFFItemClass] = true;
+								}
```

Comment:
Why do we need to do this if this is already set to true at the beginning? Should we be setting this to true/false based on if item is included?